src/runtime/virtcontainers/qemu.go

```diff
@@ -394,6 +394,23 @@ func (q *qemu) createQmpSocket() ([]govmmQemu.QMPSocket, error) {
 	return sockets, nil
 }
 
+func (q *qemu) buildInitdataDevice(devices []govmmQemu.Device, InitdataImage string) []govmmQemu.Device {
+	device := govmmQemu.BlockDevice{
```

Comment:
I may have missed some of the earlier conversation but did we consider `-fw_cfg` as an option for QEMU?

With `-fw_cfg initdata,file=hello_td.json` I can see:
```
root@tdx-guest:~# cat /sys/firmware/qemu_fw_cfg/by_name/initdata/raw 
{
	"message": "Hello, TD!",
	"action": "OK"
}
```

Reply:
OMG. This seems pretty cool and easier. 


I am not an expert in Qemu. At first, we thought CoCo disabled virt-fs, so we considered using devices to pass it. If this configuration function can be used in CoCo, it will be very convenient.

Some questions, 
1. who defines `by_name`?
2. Can it be a const in the guest side?
3. In CoCo, will `fw_cfg` introduce any security issues?

If possible, do you mind we change to this way in following PRs. Together with https://github.com/kata-containers/kata-containers/issues/10980, thus we can then decouple many of the tasks in kata-agent into a single and simple component.

Reply:
Initdata should work for all hypervisors. Now, I'm not entirely sure how we would get the device into a peer pod, but it's not impossible. `fw_cfg` will only work on qemu, though.

Reply:
@burgerdev Yes. This PR will only care about bare metal initdata implementation. For peerpod, it will be delivered by [CSP's IMDS API](https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/pkg/userdata/provision.go#L286-L314) I think. The two ways only share the annotation part. So `fw_cfg` would do nothing to remote hypervisor

Reply:
One reason why we wanted a different initdata implementation is that IMDS data often has drastic size limits, making it unsuitable for delivering policies. But peer pods may have been the wrong argument, should have said CLH.

Reply:
I have no strong opinion on `-fw_cfg` usage. I was mainly curious if it was already seen as a no-go option earlier. To me it looks simple and would avoid having to vendor 3rd party deps with 200+ new files at least in the initial implementation. How initdata gets into the guest is hypervisor specific as seen in this PR but a block dev is more future proof since it keeps the guest initdata paths consistent.

Reply:
> One reason why we wanted a different initdata implementation is that IMDS data often has drastic size limits, making it unsuitable for delivering policies. But peer pods may have been the wrong argument, should have said CLH.

Yes indeed, the AKS coco product is using Cloud Hypervisor and would need to have a different implementation for init-data if we pick a qemu-specific transport. cc @sprt